engine/crates/integration-tests/tests/federation/basic/errors.rs

```diff
@@ -95,3 +95,56 @@ fn simple_error() {
     }
     "###);
 }
+
+#[test]
+fn null_entity_with_error() {
+    let gateway = FederationGatewayWithoutIO::new(
+        SCHEMA,
+        r#"
+        query ExampleQuery {
+            me {
+                id
+                username
+                reviews {
+                    body
+                }
+            }
+        }
+        "#,
+        &[
+            json!({"data":{"me":{"id":"1234","username":"Me"}}}),
+            json!({"data":{"_entities":[null]}, "errors": [{"message":"I'm broken!", "path": ["_entities", 0, "body"]}]}),
+        ],
+    );
+    let response = integration_tests::runtime().block_on(gateway.unchecked_execute());
+
+    let json = serde_json::from_slice::<serde_json::Value>(&response.bytes).unwrap();
+    insta::assert_json_snapshot!(json, @r###"
+    {
+      "data": null,
+      "errors": [
+        {
+          "message": "Upstream response error: Missing required field named 'reviews'",
```

Comment:
Not necessarily one for this PR, but `Error decoding response from upstream` might be clearer than `Upstream response error`?

"Upstream response error" is a bit ambiguous: is it a response error coming from the upstream or an error with the upstream response...?